localstack/services/ses/provider.py

```diff
@@ -310,3 +345,96 @@ def clone_receipt_rule_set(
             backend.create_receipt_rule(rule_set_name, rule)
 
         return CloneReceiptRuleSetResponse()
+
+
+class SNSClient(Protocol):
+    def publish(self, TopicArn: str, Message: str, Subject: Optional[str] = None):
+        ...
+
+
+class SNSEmitter:
+    def __init__(
+        self,
+        context: RequestContext,
+        client: Optional[SNSClient] = None,
+    ):
+        self.context = context
+
+    def emit_create_configuration_set_event_destination_test_message(
+        self, sns_topic_arn: str
+    ) -> None:
+        client = self._client_for_topic(sns_topic_arn)
+        client.publish(
+            TopicArn=sns_topic_arn,
+            Message="Successfully validated SNS topic for Amazon SES event publishing.",
+        )
+
+    def emit_send_event(
+        self, sender_email: Address, destination_addresses: AddressList, sns_topic_arn: str
+    ):
+        now = datetime.now(tz=timezone.utc)
+
+        event_payload = {
+            "eventType": "Send",
+            "mail": {
+                "timestamp": now.isoformat(),
+                "source": sender_email,
+                "sourceArn": f"arn:aws:ses:{self.context.region}:{self.context.account_id}:identity/{sender_email}",
+                "sendingAccountId": self.context.account_id,
+                "destination": destination_addresses,
+            },
+            "send": {},
+        }
+        client = self._client_for_topic(sns_topic_arn)
+        client.publish(
+            TopicArn=sns_topic_arn,
+            Message=json.dumps(event_payload),
+            Subject="Amazon SES Email Event Notification",
+        )
+
+    def emit_delivery_event(
+        self, sender_email: Address, destination_addresses: AddressList, sns_topic_arn: str
+    ):
+        now = datetime.now(tz=timezone.utc)
+
+        event_payload = {
+            "eventType": "Delivery",
+            "mail": {
+                "timestamp": now.isoformat(),
+                "source": sender_email,
+                "sourceArn": f"arn:aws:ses:{self.context.region}:{self.context.account_id}:identity/{sender_email}",
+                "sendingAccountId": self.context.account_id,
+                "destination": destination_addresses,
+            },
+            "delivery": {
+                "recipients": destination_addresses,
+                "timestamp": now.isoformat(),
+            },
+        }
+        client = self._client_for_topic(sns_topic_arn)
+        client.publish(
+            TopicArn=sns_topic_arn,
+            Message=json.dumps(event_payload),
+            Subject="Amazon SES Email Event Notification",
+        )
+
+    @staticmethod
+    def _sender_email(request: SendEmailRequest) -> str:
+        return request["Source"]
+
+    @staticmethod
+    def _destination_addresses(request: SendEmailRequest) -> AddressList:
+        return request["Destination"]["ToAddresses"]
```

Comment:
nit: are those still in use anywhere since the changes? This is not blocking and can be removed in a follow-up PR!

Reply:
Thanks, yeah I was going to consider that (but TBH I had forgotten! :facepalm: ) I will certainly add this to an upcoming PR since proper error handling is part of the AWS contract :+1:

Reply:
Yeah that's for the future! :tada: